uni/lib/model/providers/session_provider.dart

```diff
@@ -54,9 +55,18 @@ class SessionProvider extends StateProviderNotifier {
 
         updateStatus(RequestStatus.successful);
       } else {
+        //Check if password expired
+        final String responseHtml =
+            await NetworkRouter.loginInSigarra(username, password, faculties);
+        if (isPasswordExpired(responseHtml)) {
+          action.completeError(ExpiredCredentialsException());
+        }
+        action.completeError(WrongCredentialsException());
```

Comment:
Doesn't the second `completeError` override the first? Even if not, can you put an else here to make things clearer?

Reply:
I think completeError behave here like a return statement, but I will place an else